pkg/tsdb/loki/loki.go

```diff
@@ -166,12 +203,72 @@ func parseResponse(value *loghttp.QueryResponse, query *lokiQuery) (data.Frames,
 		frames = append(frames, frame)
 	}
 
-	return frames, nil
+	return frames
+}
+
+func lokiStreamsToDataFrames(streams loghttp.Streams, query *lokiQuery) data.Frames {
+	frames := data.Frames{}
+
+	for _, v := range streams {
+		name := formatLegend(v.Labels, query)
```

Comment:
ideally the `formatLegend` stuff is not baked into the bones of parsing response.  If the response>>DataFrame conversion depends on the query, we can end up with difficult to understand implications.

Better (i think) is:
1. make request
2. convert request to DataFrame(s)
3. apply any changes to the frame based on query

this way we have a clear idea what special query based changes happen in 3 rather than wrapping that into the parsing

Reply:
good idea, it's done that way in https://github.com/grafana/grafana/pull/45653 .

(the problem with doing the frame-tuning in javascript is that we need to do some of it for alerting-queries too, so in the alerting-notification-text things get formatted the right way.)

Reply:
for now i modified this PR to be done the way you recommended.